### DISCUSSION TITLE
这个 markdown 的效果是不是会更好一些？

### DISCUSSION THREAD

**Discussion by yhm-amber** [gantrol/Bing-Chat-Saver#21] [Ideas]
示例： https://mirror.xyz/0x37Da98E4b1d562f002dF022D91E625A8b9fCc0e4/ldS-h2sZerigBDU31GslWS6pId_J0YGJx3aMjfOR0SE

要点：

- new bing 上用户发言都会被变成一行，所以我就用一个 `> ` 给弄成了引用效果。也有点以前知乎一些回答的风格：引用一下问题再回答。
- 系统提示我用的是列表。因为看到对号想到的是 `- [x] 内容` 这么个格式，所以对号也算列表……
- 回答内容是直接点的右上角三个点复制，这个应该会比目前的实现更简洁一些，而且信息也更全：并不是所有在 ref 列表里的东西都能在回答正文被引用，但肯定是有相关性的，所以我倾向于保存所有引用。

这是三个点按钮的位置：

![image](https://user-images.githubusercontent.com/103625580/227248269-8eba5c1e-01a3-4044-ae03-45a7fb428739.png)

这是一个格式比较复杂的回答案例，有代码块有链接：

![image](https://user-images.githubusercontent.com/103625580/227248550-4334a3cf-84d6-476c-a424-918507c0ea95.png)

然后这是点它的三个点复制所得到的内容：

~~~ md
在 TypeScript 中，您可以像在 JavaScript 中一样使用 `JSON.parse()` 函数来解析 JSON 字符串。下面是一个简单的示例：

```typescript
interface MyObj {
    myString: string;
    myNumber: number;
}

let obj: MyObj = JSON.parse('{ "myString": "string", "myNumber": 4 }');
console.log(obj.myString);
console.log(obj.myNumber);
```

这段代码定义了一个名为 `MyObj` 的接口，它具有两个属性：`myString` 和 `myNumber`。然后，我们使用 `JSON.parse()` 函数将一个 JSON 字符串解析为一个 `MyObj` 类型的对象，并将其存储在变量 `obj` 中。最后，我们使用 `console.log()` 函数输出对象的两个属性。

您可以参考这个 [Delft Stack](https://www.delftstack.com/zh/howto/typescript/parse-json-string-in-typescript/) 上的文章来了解更多信息。

源: 与必应的对话， 2023/3/23(1) 在 TypeScript 中解析 JSON 字符串 | D栈 - Delft Stack. https://www.delftstack.com/zh/howto/typescript/parse-json-string-in-typescript/ 访问时间 2023/3/23.
(2) 如何解析Typescript中的JSON字符串. https://qastack.cn/programming/38688822/how-to-parse-json-string-in-typescript 访问时间 2023/3/23.
(3) 如何在TS中声明一个类型，使得JSON.parse可以推断返回 .... https://www.zhihu.com/question/470163766 访问时间 2023/3/23.
(4) TypeScript 中 JSON 转 Class - 掘金. https://juejin.cn/post/7026762634993074206 访问时间 2023/3/23.
(5) TypeScript 中的Json自动解析并组装为类对象_candyguy242的 .... https://blog.csdn.net/candyguy242/article/details/84636877 访问时间 2023/3/23.
~~~

……它在回答中还是有可能会插入超链接的，所以现在的把参考直接变成超链接的 md 排版方案可能就不太优雅了，并且不排除发生冲突的可能（比如它如果对一段有着超链接的文本附上了参考链接……）。

以上，我的建议。

三个点复制出来的内容，唯一不够雅观的地方就在于，那个 `源: 与必应的对话` 和第一个参考被弄在了一行上；但我觉得也无伤大雅，反正信息都有，且也并不特别影响阅读，而且这也应该算是 new bing 界面侧的 bug ，等它修它的就行了。（是的这是个但书 ……）

**Comment by gantrol** [comment]
感谢反馈。如果有精力做 txt 格式，我会直接按必应这种格式来。

这种格式也有一定问题，比如正文引用的“1、2、3”是没有链接的，而且是很小的上标。如果回答再多几段，找链接会挺麻烦

做 Markdown 格式时也有考虑过这个，跟现在这种方案各有优劣

理想状态是将所有对话的引用变成 Markdown 脚注放在最后，不过这个要考虑的情况太多了

其他的有空再看

**Reply by yhm-amber** [reply]
引用多的时候确实找起来麻烦。

根据我的观察，一般来说，它在回答时都会尽力总结答案，对于明确要我看什么什么网页的情况则会专门给我弄超链接，然后下面 ref 会给出它搜的东西，类似于在不联网 chatGPT 那里要主动事先投喂的内容在这里它给你表明出来自己都被喂了些啥 …… 然后投喂内容和对投喂内容的总结不见得总能对应，会有投喂很多内容的时候但不见得回复中都能有办法标记上数字。

（当然，这个结论也有局限 —— 因为这只是我的观察 …… 所以你的考虑也不是没道理。）

我之前手动保存 chatGPT 的时候，用的方法就是把所有发言全搞到引用块里头。这里没弄因为觉得有点麻烦，所以对于系统通知性质的部分只是做了一个视觉效果上的模拟，而 ref 部分完全按照三个点复制给的效果来了。

--------

另外就是，这个三个点点复制后（这样得到的应该就是你说的“必应这种格式”），得到的其实并不是 TXT 文本，而是 Markdown 。它应该是只有在 ref 方面没有好好给弄格式，但回答部分该有的效果都会有，比如代码块或者超链接。

**Comment by gantrol** [comment]
微调了导出markdown的格式：https://github.com/gantrol/Bing-Chat-Saver/releases/tag/v0.2.2
